app/src/main/java/com/h/pixeldroid/PostCreationActivity.kt

```diff
@@ -77,31 +82,37 @@ class PostCreationActivity : AppCompatActivity() {
         accessToken = user?.accessToken.orEmpty()
         pixelfedAPI = PixelfedAPI.create(domain)
 
-        // check if the picture is alright
-        // TODO
-
-        // edit the picture
-        // TODO
+        //upload the picture and display progress while doing so
+        upload()
 
-        // get the description and send the post to PixelFed
+        // get the description and send the post
         findViewById<Button>(R.id.post_creation_send_button).setOnClickListener {
-            if (setDescription()) upload()
+            if (setDescription() && listOfIds.isNotEmpty()) post()
         }
+        
+        // Button to retry image upload when it fails
+        findViewById<Button>(R.id.retry_upload_button).setOnClickListener {
+            upload_error.visibility = GONE
+            upload()
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        //delete the temporary image
+        image.delete()
     }
 
     private fun saveImage(imageUri: Uri) {
-        val timeStamp: String = SimpleDateFormat("yyyyMMdd_HHmmss").format(Date())
-        val fileName = "PixelDroid_$timeStamp.png"
```

Comment:
should keep unique name

Reply:
lol yes, thanks

Reply:
no, actually this is done by the createTempfile already.